[bitnami/zookeeper] Enable JSON format structured logging

janhoy opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

### Description of the change

* Adds custom `logback.xml` log configuration file for zookeeper.
* Change log level not through xml patching but through system property `zookeeper.log.level`
* Change logger appender through sysprop `zookeeper.log.appender` or env.var `ZOO_LOG_APPENDER`
* Enable JSON format structured logging through sysprop `zookeeper.log.format=json` or env.var `ZOO_LOG_FORMAT=json`
* New script `libmaven.sh` that downloads required maven jar artifacts

### Benefits

See issue. Greater control over logging.

### Possible drawbacks

* Additional weight from added jar files
* Harder to keep jar dependencies up to date when they are not part of maven build

### Applicable issues

- fixes #51921

### Additional information

This is a WIP, only added for 3.9.x. I'm sure there is some smarter way to add the same for 3.8 than to copy/paste the same files around - is there some script that generates each version?

Instead of adding new `COPY` and `RUN` commands to Dockerfile I included the xml file in `rootfs` and the jar download commands to `postunpack.sh`, not sure if this is best practice?

janhoy committed:
Enable JSON format structured logging

janhoy committed:
Remove libzookeeper ZOO_LOG_LEVEL patching

Signed-off-by: Jan Høydahl <janhoy@users.noreply.github.com>

carrodher commented:
Thank you for initiating this pull request. We appreciate your effort. Just a friendly reminder that it's important to sign your commits. Adding your signature certifies that you either authored the patch or have the necessary rights to contribute the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

janhoy committed:
Add docs for ZOO_LOG_FORMAT
Fix ZOO_LOG_LEVEL docs to use the logback logging levels, not the old log4j ones.

Signed-off-by: Jan Høydahl <janhoy@users.noreply.github.com>

janhoy commented:
Looks like you're waiting for me to sign my commits before having a look. So just bumping here for the stale-bot to notice.

javsalgar commented:
Hi!

Yes, before passing it to the reviewers, we need you to fix the DCO issue.
